include/aspect/geometry_model/ellipsoidal_chunk.h

```diff
@@ -0,0 +1,268 @@
+/*
+  Copyright (C) 2011 - 2015 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file doc/COPYING.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#ifndef __aspect__geometry_model_ellipsoidal_chunk_h
+#define __aspect__geometry_model_ellipsoidal_chunk_h
+
+#include <aspect/geometry_model/interface.h>
+#include <deal.II/grid/manifold.h>
+
+/**
+ * This geometry model implements an (3d) ellipsoidal chunk geometry where two of the axis have
+ * the same length. The ellipsoidal chunk can be non-coordinate parallel part of the ellipsoid.
+ * @author This plugin is a joined effort of Menno Fraters, D Sarah Stamps and Wolfgang Bangerth
+ */
+
+namespace aspect
+{
+  namespace GeometryModel
+  {
+    using namespace dealii;
+
+    /**
+     * A class that describes a geometry for an ellipsoid such as the WGS84 model of the earth.
+     */
+    template <int dim>
+    class EllipsoidalChunk : public Interface<dim>
+    {
+      public:
+        /**
+         * A class which describes the manifold.
+         */
+        class EllipsoidalChunkGeometry : public ChartManifold<dim,3,3>
+        {
+          public:
+
+            void
+            set_initial_values(double para_semi_major_axis_a,
+                               double para_eccentricity,
+                               double para_semi_minor_axis_b,
+                               double para_rot_para_to_para_angle,
+                               double para_para_to_rect_angle,
+                               double para_rotation_longitude,
+                               double para_rotation_latitude,
+                               double para_bottom_depth);
+
+            virtual
+            Point<3>
+            pull_back(const Point<3> &space_point) const;
+
+            virtual
+            Point<2>
+            pull_back(const Point<2> &space_point) const;
+
+            virtual
+            Point<3>
+            push_forward(const Point<3> &chart_point) const;
+
+            Point<dim>
+            rotate_rectangle_to_parallelogram(const Point<dim> &x) const;
+
+            Point<dim>
+            rotate_parallelogram(const Point<dim> &x) const;
+
+
+
+          private:
+            double semi_major_axis_a = -1;
```

Comment:
Initializing member variables like this has unfortunately only been possible since C++11. Can you add a constructor to this class that does the initialization of these member variables?